modules/dreamview/backend/teleop/teleop.cc

```diff
@@ -104,23 +132,82 @@ void TeleopService::RegisterMessageHandlers() {
       });
 }
 
-void TeleopService::UpdateModemInfo(
-    const std::shared_ptr<ModemInfo> &modem_info) {
-  // TODO simplify data and only send necessary info for display
-  if (modem_info->has_provider() && modem_info->has_technology()) {
-    boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
-    teleop_status_["modems"][modem_info->provider()] = modem_info->technology();
-  }
-}
 
 void TeleopService::SendStatus(WebSocketHandler::Connection *conn) {
   std::string to_send;
   {
     boost::shared_lock<boost::shared_mutex> reader_lock(mutex_);
     to_send = teleop_status_.dump();
+    AINFO << "status: " << to_send;
   }
   websocket_->SendData(conn, to_send);
 }
 
+
+void TeleopService::UpdateModem(unsigned int index,
+    const std::shared_ptr<ModemInfo> &modem_info) {
+
+    boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
```

Comment:
please move this lock to line 167

Reply:
fixed [here](https://github.com/ApolloAuto/apollo/pull/9963/commits/c316aa8253165a9ced4524995c176569ca781045)